great_expectations/rule_based_profiler/domain_builder/cardinality_checker.py

```diff
@@ -0,0 +1,211 @@
+    REL_50 = RelativeCardinalityLimit("REL_50", 0.50)
+    REL_75 = RelativeCardinalityLimit("REL_75", 0.75)
+    ONE_PCT = RelativeCardinalityLimit("ONE_PCT", 0.01)
+    TEN_PCT = RelativeCardinalityLimit("TEN_PCT", 0.10)
+
+
+class CardinalityChecker:
+    """Handles cardinality checking given cardinality limit mode and measured value.
+
+    This class also validates cardinality limit settings and converts from
+    various types of settings. You can choose one of the attributes listed
+    below to create an instance.
+
+    Attributes:
+        limit_mode: CardinalityLimitMode or string name of the mode
+            defining the maximum allowable cardinality.
+        max_unique_values: number of max unique rows for a custom
+            cardinality limit.
+        max_proportion_unique: proportion of unique values for a
+            custom cardinality limit.
+    """
+
+    SUPPORTED_CARDINALITY_LIMIT_MODE_CLASSES: Tuple[
+        Union[AbsoluteCardinalityLimit, RelativeCardinalityLimit]
+    ] = (
+        AbsoluteCardinalityLimit,
+        RelativeCardinalityLimit,
+    )
+    SUPPORTED_CARDINALITY_LIMIT_MODE_STRINGS: Tuple[str] = (
+        mode.name for mode in CardinalityLimitMode
+    )
+
+    def __init__(
+        self,
+        limit_mode: Optional[Union[CardinalityLimitMode, str]] = None,
+        max_unique_values: Optional[int] = None,
+        max_proportion_unique: Optional[Union[int, float]] = None,
+    ):
+        self.supported_limit_mode_class_names = (
+            mode.__name__ for mode in self.SUPPORTED_CARDINALITY_LIMIT_MODE_CLASSES
+        )
+
+        self._limit_mode = self._convert_to_cardinality_mode(
+            limit_mode=limit_mode,
+            max_unique_values=max_unique_values,
+            max_proportion_unique=max_proportion_unique,
+        )
+
+    @property
+    def limit_mode(self) -> CardinalityLimitMode:
+        return self._limit_mode
+
+    def cardinality_within_limit(self, metric_value: Union[int, float]) -> bool:
+        """Determine if the cardinality is within configured limit.
+
+        The metric_value supplied should be either a proportion of unique values
+        or number of unique values based on the configured cardinality limit.
+
+        Args:
+            metric_value: int if number of unique values, float if proportion
+                of unique values.
+
+        Returns:
+            boolean of whether the cardinality is within the configured limit
+        """
+        self._validate_metric_value(metric_value=metric_value)
+        if isinstance(self._limit_mode, AbsoluteCardinalityLimit):
+            return metric_value <= self._limit_mode.max_unique_values
+        elif isinstance(self._limit_mode, RelativeCardinalityLimit):
+            return metric_value <= self._limit_mode.max_proportion_unique
+
+    @staticmethod
+    def _validate_metric_value(metric_value: Union[int, float]) -> None:
+        if not isinstance(metric_value, (int, float)):
```

Comment:
@anthonyburdi We have an `is_numeric()` checker in greatexpectations/util.py -- perhaps it might work here?  Thanks!

Reply:
Thank you for the reminder about our util methods @alexsherstinsky ! In this case it's not exactly equivalent, here I'm just checking the type of `metric_value`. In `is_numeric()` it tries to coerce the value and check for errors - so for example a string would be considered numeric e.g. `is_numeric("1") == True` whereas here we just want to ensure an int or float type. If we want to handle numeric strings we would do the conversion outside of the responsibility of the `CardinalityChecker`.

Interestingly, we turned `Union[int, float]` to `float` based on pep484 in another comment, but `isinstance()` behaves differently e.g. `isinstance(1, float) == False`

Reply:
@anthonyburdi There is a chance that we can get some help for this code segment by merging #4282 first.  Thanks!